src/train.js

```diff
@@ -193,7 +206,20 @@ function updateProgress(ratio, bestSolution, bestCost, successesOrFailures) {
  * Draw and outfit the Train page.
  */
 async function initPage(document) {
-    document.getElementById('train').onclick = trainOnTabs;
+    // Find elements once.
+    g_progressBar = document.getElementById('progress');
+    g_coeffs_div = document.getElementById('coeffs');
+    g_accuracy_div = document.getElementById('accuracy');
+    g_goodBadDiv = document.getElementById('goodBad');
+
+    // Initialise elements to a known state.
+    empty(g_coeffs_div);
+    empty(g_accuracy_div);
+    empty(g_goodBadDiv);
+
+    // Create a text node in coeffs and accuracy once, rather than on each update.
+    g_coeffs_div.appendChild(document.createTextNode(''));
```

Comment:
Aha. Couldn't we more briefly just stick a space in the HTML, perhaps with a comment?

Reply:
we could, but i think that's more fragile, and it keeps the initialisation near the code where the structure of the element matters.

Reply:
That's a valid argument (hence the comment about comments). Too bad it's so long in DOM-API spelling, though.